File: BinaryTree/java/maxPathSum.java
class MaxPathSum {
  public static int maxPathSum(TreeNode root) {
    if (root == null) {
      return Integer.MIN_VALUE;
    }
    if (root.left == null && root.right == null) {
      return root.val;
    }
    return root.val + Math.max(maxPathSum(root.left), maxPathSum(root.right));
  }

  public static void run() {
    // intentionally empty
  }
}

File: BinaryTree/csharp/MaxPathSum.cs
using System;

namespace BinaryTreeSolutions;

public static class MaxPathSum
{
    public static int Solve(TreeNode? root)
    {
        if (root is null)
        {
            return int.MinValue;
        }

        if (root.Left is null && root.Right is null)
        {
            return root.Val;
        }

        return root.Val + Math.Max(Solve(root.Left), Solve(root.Right));
    }
}
